app/src/main/java/io/neurolab/main/TestModeActivity.java

```diff
@@ -0,0 +1,158 @@
+package io.neurolab.main;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.hardware.usb.UsbDevice;
+import android.hardware.usb.UsbDeviceConnection;
+import android.hardware.usb.UsbManager;
+import android.os.Bundle;
+import android.support.design.widget.FloatingActionButton;
+import android.support.design.widget.Snackbar;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.Toolbar;
+import android.util.Log;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Spinner;
+import android.widget.TextView;
+
+import com.felhr.usbserial.UsbSerialDevice;
+import com.felhr.usbserial.UsbSerialInterface;
+
+import java.io.UnsupportedEncodingException;
+
+import io.neurolab.R;
+
+public class TestModeActivity extends AppCompatActivity implements AdapterView.OnItemSelectedListener {
+    private DeviceConnector deviceConnector;
+    private Button readBtn;
+    private Button writeBtn;
+    private EditText editText;
+    private TextView displayView;
+    private Spinner baudRateSpinner;
+
+    private UsbManager usbManager;
+    private UsbDevice device;
+    private UsbSerialDevice serialPort;
+    private UsbDeviceConnection connection;
+    private int baudRate = 9600;
+
+    private final String ACTION_USB_PERMISSION = "io.neurolab.main.USB_PERMISSION";
+
+    UsbSerialInterface.UsbReadCallback callBack = new UsbSerialInterface.UsbReadCallback() { //Defining a Callback which triggers whenever data is read.
```

Comment:
Make this private and remove the generic package protected scoping. Work with getters and setters (though I think won't be required in this case).